recipes/agent_linux.rb

```diff
@@ -25,90 +25,135 @@
-end
+  # Create and manage the agent directory
+  directory node['newrelic_infra']['agent']['directory']['path'] do
+    owner node['newrelic_infra']['user']['name']
+    group node['newrelic_infra']['group']['name']
+    mode node['newrelic_infra']['agent']['directory']['mode']
+  end
 
-# Build the New Relic infrastructure agent configuration
-file node['newrelic_infra']['agent']['flags']['config'] do
-  content(lazy do
-    YAML.dump(
-      node['newrelic_infra']['config'].to_h.deep_stringify.delete_blank
-    )
-  end)
-  owner node['newrelic_infra']['user']['name']
-  group node['newrelic_infra']['group']['name']
-  mode  '0640'
-  sensitive true
-  notifies :restart, 'poise_service[newrelic-infra]'
-end
+  # Build the New Relic infrastructure agent configuration
+  file node['newrelic_infra']['agent']['flags']['config'] do
+    content(lazy do YAML.dump(
+        node['newrelic_infra']['config'].to_h.deep_stringify.delete_blank
+      )
+    end)
+    owner node['newrelic_infra']['user']['name']
+    group node['newrelic_infra']['group']['name']
+    mode  '0640'
+    sensitive true
+    notifies :restart, 'poise_service[newrelic-infra]'
+  end
+
+  # Enable and start the agent as a service on the node with any available
+  # CLI options
+  poise_service 'newrelic-infra' do
+    # TODO: Figure out how to run as a service account.
+    # user node['newrelic_infra']['user']['name']
+    command '/usr/bin/newrelic-infra ' <<
+            NewRelicInfra.generate_flags(node['newrelic_infra']['agent']['flags'])
+    options(:systemd,
+            template: 'newrelic-infra:default/systemd.service.erb',
+            after: %w(syslog.target network.target))
+  end
+when 'tarball'
+  node['newrelic_infra']['tarball'].tap do |conf|
+    remote_file "/opt/linux_#{conf['version']}_#{conf['architecture']}.tar.gz" do
+      source "https://download.newrelic.com/infrastructure_agent/binaries/linux/#{conf['architecture']}/newrelic-infra_linux_#{conf['version']}_#{conf['architecture']}.tar.gz"
+      action :create
+    end
+    directory '/opt/newrelic_infra/' do
+      action :create
+    end
+    directory "/opt/newrelic_infra/linux_#{conf['version']}_#{conf['architecture']}" do
+      action :create
+    end
 
-include_recipe 'newrelic-infra::host_integrations'
-# Enable and start the agent as a service on the node with any available
-# CLI options
-poise_service 'newrelic-infra' do
-  # TODO: Figure out how to run as a service account.
-  # user node['newrelic_infra']['user']['name']
-  command '/usr/bin/newrelic-infra ' <<
-          NewRelicInfra.generate_flags(node['newrelic_infra']['agent']['flags'])
-  options(:systemd,
-          template: 'newrelic-infra:default/systemd.service.erb',
-          after: %w(syslog.target network.target))
+    execute 'extract_newrelic_infra_tarball' do
+      command "tar -xzf /opt/linux_#{conf['version']}_#{conf['architecture']}.tar.gz -C /opt/newrelic_infra/linux_#{conf['version']}_#{conf['architecture']}/"
```

Comment:
Shouldn't be better to download the tar.gz in /tmp?

Reply:
The problem with that is that if `tmp` is cleared chef will download and install the agent again, this way we assert that if it's there the steps have been performed and there is no need to do it again.